src/server/pfs/server/http.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/net/context"
+)
+
+// HTTPPort specifies the port the server will listen on
+const HTTPPort = 652
+const apiVersion = "v1"
+
+type flushWriter struct {
+	f http.Flusher
+	w io.Writer
+}
+
+func (fw *flushWriter) Write(p []byte) (n int, err error) {
+	n, err = fw.w.Write(p)
+	if fw.f != nil {
+		fw.f.Flush()
+	}
+	return
+}
+
+// HTTPServer serves GetFile requests over HTTP
+// e.g. http://localhost:30652/v1/pfs/repos/foo/commits/b7a1923be56744f6a3f1525ec222dc3b/files/ttt.log
+type HTTPServer struct {
+	driver *driver
+}
+
+func newHTTPServer(address string, etcdAddresses []string, etcdPrefix string, cacheBytes int64) (*HTTPServer, error) {
+	d, err := newDriver(address, etcdAddresses, etcdPrefix, cacheBytes)
+	if err != nil {
+		return nil, err
+	}
+	return &HTTPServer{d}, nil
+}
+
+// Start initiates the server's ListenAndServe method
+func (s *HTTPServer) Start() error {
+	router := httprouter.New()
+	router.GET(fmt.Sprintf("/%v/pfs/repos/:repoName/commits/:commitID/files/*filePath", apiVersion), s.getFileHandler)
+	// Serves a request like:
+	// http://localhost:30652/v1/pfs/repos/foo/commits/b7a1923be56744f6a3f1525ec222dc3b/files/ttt.log
+	return http.ListenAndServe(fmt.Sprintf(":%v", HTTPPort), router)
+}
```

Comment:
Works for me. And then `pachd/main.go` will specify the port to run the server on as well. One question about ports --- if the container port is `80` do we want the nodeport to be `30080` ? Because the dash UI already uses that node port.